test_tools/sigma_verify_rules.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A tool to test sigma rules.
+
+This tool can be used to verify your rules before running an analyzer.
+It also does not require you to have a full blown Timesketch instance.
+Example way of running the tool:
+  $ sigma_verify_rules.py --config_file ../data/sigma_config.yaml ../data/linux/
+"""
+
+
+import logging
+import os
+import argparse
+import sys
+import codecs
+import sigma.parser.exceptions
+import sigma.configuration as sigma_configuration
+
+from sigma.backends import elasticsearch as sigma_elasticsearch
+from sigma.parser import collection as sigma_collection
+
+logger = logging.getLogger('timesketch.test_tool.sigma-verify')
+logging.basicConfig(level=os.environ.get('LOGLEVEL', 'ERROR'))
+
+RULE_EXTENSIONS = ('yml', 'yaml')
+
+def get_codepath():
+    """Return the absolute path to where the tool is run from."""
+    # TODO: move this function to a library as it is duplicate to WebUI
+
+    path = __file__
+    if path.startswith(os.path.sep):
+        return path
+
+    dirname = os.path.dirname(path)
+    for sys_path in sys.path:
+        if sys_path.endswith(dirname):
+            return sys_path
+    return dirname
+
+def verify_rules_file(rule_file_path, sigma_config, sigma_backend):
+    """Verifies a given file path contains a valid sigma rule.
+
+        Args:
+            rule_file_path: the path to the rules.
+            sigma_config: config to use
+            sigma_backend: A sigma.backends instance
+
+        Raises:
+            None
+
+        Returns:
+            true: rule_file_path contains a valid sigma rule
+            false: rule_file_path does not contain a valid sigma rule
+    """
+    logger.debug('[sigma] Reading rules from {0:s}'.format(
+        rule_file_path))
+
+    try:
+        path, rule_filename = os.path.split(rule_file_path)
+
+        with codecs.open(rule_file_path, 'r', encoding="utf-8") as rule_file:
```

Comment:
be consistent with the use of '' vs ""

encoding='utf-8'

Reply:
Fixed